homeassistant/components/plugwise/climate.py

```diff
@@ -124,15 +139,18 @@ def extra_state_attributes(self) -> Mapping[str, Any] | None:
     @plugwise_command
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        if ((temperature := kwargs.get(ATTR_TEMPERATURE)) is None) or (
-            self._attr_max_temp < temperature < self._attr_min_temp
+        if ((temperature := kwargs.get(ATTR_TEMPERATURE)) is None) or not (
+            self._attr_min_temp < temperature < self._attr_max_temp
         ):
             raise ValueError("Invalid temperature requested")
         await self.coordinator.api.set_temperature(self.device["location"], temperature)
 
     @plugwise_command
     async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set the hvac mode."""
+        if hvac_mode not in self.hvac_modes:
+            raise HomeAssistantError("Unsupported hvac_mode")
```

Comment:
Hmm not for this PR or this integration even... but the Home Assistant project should actually guard/fix this in the core services of the climate platform.

Reply:
Yeah, I've noticed that I could enter an invalid string and then there was no error. This check helps in such a case.

Do you want me to remove the check? Or leave it for now?

Reply:
No, leave it, just ignore this comment... was more a note to self 😬